pysph/examples/db_2d_buchner.py

```diff
@@ -0,0 +1,319 @@
+"""Two-dimensional dam break over a dry bed.  (30 minutes)
+
+The case is described in "B.Buchner, Green Water on Ship -
+Type Off shore Structures, Ph.D.thesis,TU Delft,
+Delft University of Technology (2002), Appendix II"
+
+"""
+
+import os
+import numpy as np
+
+from pysph.base.kernels import WendlandQuintic, QuinticSpline
+from pysph.base.utils import get_particle_array
+from pysph.solver.application import Application
+from pysph.sph.scheme import WCSPHScheme, SchemeChooser, AdamiHuAdamsScheme
+from pysph.sph.wc.edac import EDACScheme
+from pysph.sph.iisph import IISPHScheme
+from pysph.sph.equation import Group
+from pysph.sph.wc.kernel_correction import (GradientCorrectionPreStep,
+                                            GradientCorrection,
+                                            MixedKernelCorrectionPreStep)
+from pysph.sph.wc.crksph import CRKSPHPreStep, CRKSPH
+from pysph.sph.wc.gtvf import GTVFScheme
+from pysph.sph.isph.sisph import SISPHScheme
+from pysph.tools.geometry import get_2d_tank, get_2d_block
+
+fluid_column_height = 1.0
+H = fluid_column_height
+fluid_column_width = 2.0 * H
+container_height = 3.0 * H
+container_width = 5.366 * H
+nboundary_layers = 2
+nu = 0.0
+dx = 0.03
+g = 9.81
+ro = 1000.0
+vref = np.sqrt(2 * 9.81 * H)
+co = 10.0 * vref
+gamma = 7.0
+alpha = 0.1
+beta = 0.0
+B = co * co * ro / gamma
+p0 = 1000.0
+hdx = 1.3
+h = hdx * dx
+m = dx**2 * ro
+
+
+class DamBreak2D(Application):
+
+    def add_user_options(self, group):
+        corrections = ['', 'mixed-corr', 'grad-corr', 'kernel-corr', 'crksph']
+        interp_methods = ['shepard', 'sph', 'order1']
+        group.add_argument(
+            '--dx', action='store', type=float, dest='dx', default=dx,
+            help='Particle spacing.'
+        )
+        group.add_argument(
+            '--hdx', action='store', type=float, dest='hdx', default=hdx,
+            help='Specify the hdx factor where h = hdx * dx.'
+        )
+        group.add_argument(
+            "--kernel-corr", action="store", type=str, dest='kernel_corr',
+            default='', help="Type of Kernel Correction", choices=corrections
+        )
+        group.add_argument(
+            '--staggered-grid', action="store_true", dest='staggered_grid',
+            default=False, help="Use a staggered grid for particles.",
+        )
+        group.add_argument(
+            '--interp-method', action="store", type=str, dest='interp_method',
+            default='shepard', help="Specify the interpolation method.",
+            choices=interp_methods
+        )
+
+    def consume_user_options(self):
+        self.hdx = self.options.hdx
+        self.dx = self.options.dx
+        self.h = self.hdx * self.dx
+        self.kernel_corr = self.options.kernel_corr
+        self.interp_method = self.options.interp_method
+        print("Using h = %f" % self.h)
+
+    def configure_scheme(self):
```

Comment:
Hmm this all looks the same as DB 2d?  If this is all cut/pasted from there, then this is a major no-no as it is very very difficult to maintain multiple cut/pasted pieces of code. Instead try to refactor this so there is maximum code reuse.